docs: per-agent reference with icons, commands, and labels (#1017)
konflux-ci/fullsend: docs/agents/README.md

# Out-of-the-Box Agents

> Reference documentation for the agents shipped in fullsend's default harness.

Reviewer: This makes the user think that there is only one harness.

```suggestion
Reference documentation for the default agents shipped by fullsend.
```

Author: Agreed!

Author: resolved in commit 9f44170

> All agents below are enabled by default. The set of OOTB agents is defined by

Reviewer: I knew the OOTB acronym would appear. If we will be OOTB every time let's use another term rather than Out Of The Box. "Default agents" sound good, for example.

Author: Hm, do we need to account for three different categories?

* Out of the box (OOTB) agents. These are agents that are provided by fullsend.
* Default agents. These are a _subset_ of the OOTB agents. These are the agents that are enabled by default on an install. Some OOTB agents are available, but not enabled by default.
* Custom agents. These are agents introduced entirely by the user in `customized/`.

I'm not super in love with "OOTB". Default agents can work. With ^^ context in mind, would you still prefer "Default agents" in place of "Out of the box agents"?

Author: For now - ok; let's just switch to "Default Agents" everywhere instead of "Out of the Box Agents" and "OOTB". Scrub all uses of those terms and replace them with "default agents".

Author: Scrubbed OOTB → default in c9f3a6f (README, lint script, scaffold comment).

# Default Agents

Reference documentation for the default agents shipped by fullsend.
All agents below are enabled by default. The set of default agents is defined by
the YAML files in [`internal/scaffold/fullsend-repo/harness/`](../../internal/scaffold/fullsend-repo/harness/).

| Agent | Summary |
|-------|---------|
| [Triage](triage.md) | Inspects new issues and produces structured triage decisions |
| [Prioritize](prioritize.md) | Scores issues using the RICE framework for project board ranking |
| [Code](code.md) | Implements fixes and features from triaged issues |
| [Review](review.md) | Reviews pull requests for correctness, security, and intent alignment |
| [Fix](fix.md) | Addresses review feedback on open PRs |
| [Retro](retro.md) | Analyzes completed workflows and proposes system improvements |

## Customization

All agents can be customized by adding instructions and skills to your
repository. Changes to `AGENTS.md` affect every agent; skills let you tune how
a specific agent performs a specific task. See
[Customizing with AGENTS.md](../guides/user/customizing-with-agents-md.md) and
[Customizing with Skills](../guides/user/customizing-with-skills.md).

> ## Bring Your Own Agent (BYOA)

Reviewer: Please, don't use BYOA as an acronym 🤣 if Bring Your Own Agent is too long, use something shorter.

Edit: we can keep the title, but without the BYOA acronim. For example a I don't want to read "When you BYOA, place its harnes here and there" or "BYOA agents are taken very seriously", etc.

Author: Let's just call them "Custom Agents" so we have "Default Agents" and "Custom Agents".

Reviewer: I think that sounds good. I know we like acronyms and sutff, but countless time we do not know what people is talking about.

Author: Done in c9f3a6f. BYOA → Custom Agents, OOTB → default agents.
